### ISSUE TITLE
EFK: Error checking ACL when seeding

### ISSUE THREAD

**Issue by rflorenc** [openshift/origin-aggregated-logging#152]
**Issue:**
After doing a basic installation of EFK logging on a lab cluster, the following errors come up in all of the Elasticsearch pods:

[2016-05-23 15:26:50,982][ERROR][io.fabric8.elasticsearch.plugin.acl.DynamicACLFilter] [Stained Glass Scarlet] Error checking ACL when seeding
org.elasticsearch.cluster.block.ClusterBlockException: blocked by: [SERVICE_UNAVAILABLE/1/state not recovered / initialized];
...

**For further details see also:**
[(https://gist.github.com/rflorenc/18644314624ff0876b9f62a8e30ac25c)]

**Version info:**
openshift v3.2.0.44
kubernetes v1.2.0-36-g4a3f9c5
etcd 2.2.5

oc get pods -o wide
NAME                          READY     STATUS      RESTARTS   AGE       NODE
logging-deployer-0ihvw        0/1       Completed   0          2d        192.1.12.26
logging-deployer-8vlkb        0/1       Completed   0          2d        192.1.12.26
logging-deployer-98bpu        0/1       Completed   0          2d        192.1.12.26
logging-deployer-dwyno        0/1       Completed   0          2d        192.1.12.26
logging-deployer-jd6so        0/1       Completed   0          2d        192.1.12.26
logging-deployer-rpri1        0/1       Completed   0          2d        192.1.12.26
logging-deployer-xtgde        0/1       Completed   0          2d        192.1.12.26
logging-es-0egyxsqv-1-7yo6f   1/1       Running     0          2d        192.1.13.21
logging-es-0tkimh6h-1-52ejb   1/1       Running     0          2d        192.1.13.44
logging-es-1jnuwamn-1-2qgsu   1/1       Running     0          2d        192.1.12.91
logging-es-1wutikij-1-q63ed   1/1       Running     0          2d        192.1.13.30
logging-es-6mdp2lg4-1-qaqda   1/1       Running     0          2d        192.1.13.35
logging-es-6qvmrgr3-1-66jhn   1/1       Running     0          2d        192.1.12.80
logging-es-agd26le3-1-tc59s   1/1       Running     0          2d        192.1.13.16
logging-es-blvicocl-1-jygw6   1/1       Running     0          2d        192.1.13.26
logging-es-crg80ae2-1-cxzgd   1/1       Running     0          2d        192.1.13.33
logging-es-cu4hcxij-1-st9og   1/1       Running     0          2d        192.1.12.81
logging-es-ewaf2h33-1-snp2v   1/1       Running     0          2d        192.1.12.86
logging-es-ki533gnt-1-5hx7x   1/1       Running     0          2d        192.1.13.1
logging-es-m62n8qre-1-comug   1/1       Running     0          2d        192.1.13.14
logging-es-qb6hswff-1-5cosr   1/1       Running     0          2d        192.1.13.4
logging-es-qdbdftok-1-v2foz   1/1       Running     0          2d        192.1.13.51
logging-es-shm6lj2r-1-wxs1f   1/1       Running     0          2d        192.1.13.28
logging-es-sq0arlq1-1-flv71   1/1       Running     0          2d        192.1.12.96
logging-es-uvcyndsx-1-6njbd   1/1       Running     0          2d        192.1.13.17
logging-es-ve4k3a1i-1-nhuot   1/1       Running     0          2d        192.1.13.20
logging-es-y4fts5fj-1-7n9zw   1/1       Running     0          2d        192.1.12.78
logging-kibana-1-5915n        2/2       Running     0          2d        192.1.12.26

**Comment by richm** [comment]
I think this is a spurious error caused because searchguard is attempting to perform some operation that requires the entire cluster to be up and running first.  Please verify that access control is actually working once the whole cluster is available.  You can use the cluster health endpoint https://www.elastic.co/guide/en/elasticsearch/guide/1.x/cluster-health.html or the cluster stats endpoint https://www.elastic.co/guide/en/elasticsearch/guide/1.x/_cluster_stats.html to check the status.  Finally, I don't know if this issue is a duplicate of https://github.com/openshift/origin-aggregated-logging/issues/22 but it is related.

**Comment by ewolinetz** [comment]
Do we ever see ES recover/start up in the logs?  (https://github.com/openshift/origin-aggregated-logging#elasticsearch-1)

I'm less concerned with the error sourcing from io.fabric8.elasticsearch.plugin.acl.DynamicACLFilter as we would expect any operations against ES fail if the cluster is not yet fully up and accepting incoming requests.

**Comment by ewolinetz** [comment]
This is resolved in the es2.x branch

**Comment by richm** [comment]
Please add a link to the commit that fixed this issue and close the issue.

**Comment by ewolinetz** [comment]
Resolved with: https://github.com/fabric8io/openshift-elasticsearch-plugin/commit/f3c1ec175726daef2cf7ca5db1ea4546e442b865

**Comment by ebalsumgo** [comment]
I am actually seeing this exact error as we speak on OCP 3.3.1.3 when i looked at doing a DR test and took out the nodes that hosted ES nodes. Now they are trying to recover and my logs are:
[ERROR][io.fabric8.elasticsearch.plugin.acl.DynamicACLFilter] [Heinrich Zemo] Error checking ACL when seeding
org.elasticsearch.cluster.block.ClusterBlockException: blocked by: [SERVICE_UNAVAILABLE/1/state not recovered / initialized];

Filled with the above and the cluster never comes back up. The rest of the stack recovered but without ES, it doesnt matter. Any suggestions?
